fix: add sp creation to prerequirements (#27)
Azure-Samples/enterprise-data-sharing: 1-data-marketplace/README.md

| Ama | offerTier |  Tier offering | standard |
| Ama | crossTenant |  Data Sharing solution to work across tenants | tue or false |
| Ama | name |  Name of the Managed application | E.g: DataMarketPlace_Internal |
| Ama | tenantId |  Tenant ID | Fill the Tenant ID of the subscription |

The prerequisites.sh script is responsible for the automation of:

- 4 Resource Groups as defined in the config file (Definition RG, Managed App RG, Managed RG and Co-Manged RG)
- 3 Entra ID groups as defined in the config file and replace just-in-time created Entra IDs in the config file
- 2 Application Registrations in Entra ID: analytics-sp and subowner_sp replace just-in-time created Entra IDs in the config file
- Signed-in user will be added to the KeyVault Entra ID and DevOps Entra ID groups
- analytics-sp app registration will be added to the synapseSqlAdmin Entra ID Group
- subowner-sp app registration will be added to the synapseSqlAdmin Entra ID Group
- Reader RBAC role to analytics-sp
- Owner Subscription RBAC role to subowner-sp

To run the script, run the following commands:

```bash
cd 1-data-marketplace/scripts
bash prerequirements.sh
```

### Graph Permissions

> Additionally, the analytics-sp needs to be granted to the following Graph API application permissions.

Reviewer: ```suggestion
Furthermore, the analytics-sp requires authorization for the specified Graph API application permissions. To accomplish this, assign the General Administrator role to the sp_graph service principal created during the prerequirements.sh process within your Entra ID manually.
```

In order to do that, manually give the sp_graph service principal created in the prerequirements.sh step General Administrator in your Entra ID.

> In order tu run the automation for the Graph permissions follow the steps below:

Reviewer: ```suggestion
In order to run the automation for the Graph permissions follow the steps below:
```

> 1 - Navigate to 1-data-marketpace/scripts

Reviewer: ```suggestion
1 - Navigate to [Scripts](/enterprise-data-sharing/1-data-marketplace/scripts) 

```

> 2 - Run pip install -r requirements.txt OR conda env create -f requirementmsgraphpython.yml

Reviewer: ```suggestion

2 - Run pip install -r requirements.txt 
    OR 
    conda env create -f requirementmsgraphpython.yml
    
```

> 3 - Run add-graph-permissions.py. This script will be responsible to add the following permissions to the sp_graph service principal:

Reviewer: ```suggestion

3 - Run [add-graph-permissions.py](/enterprise-data-sharing/1-data-marketplace/scripts/add-graph-permissions.py) . This script will be responsible to add the following permissions to the sp_graph service principal:

```

>     - Applications.Read.All

Reviewer: ```suggestion

    - Applications.Read.All
    
```

Author: I get linting errors if I add the lines in between - can't change this one.

>     - Group.Read.All

Reviewer: ```suggestion

    - Group.Read.All
    
```

Author: I get linting errors if I add the lines in between - can't change this one.

>     - User.Read.All

Reviewer: ```suggestion

    - User.Read.All
    
```

Author: I get linting errors if I add the lines in between - can't change this one.

>     - GroupMember.Read.All

Reviewer: ```suggestion

    - GroupMember.Read.All
    
```

Author: Same as before
